### DISCUSSION TITLE
Can I provide matrix to fitz.utils.get_text?

### DISCUSSION THREAD

**Discussion by ivanstepanovftw** [pymupdf/PyMuPDF#2850] [Looking for help] [answered]
Similarly to `fitz.utils.get_pixmap`, where I can provide `matrix` or `dpi` argument:
```
page_pixmap = fitz.utils.get_pixmap(
    page=page,
    matrix=fitz.Matrix(5, 5),  # 360 dpi
    clip=span["bbox"],
    colorspace="RGB",
    alpha=False,
)
```
I want to use the `matrix` (or `dpi`) in the `fitz.utils.get_text` function. This is to ensure that the pixel units are scaled consistently with the page pixmap.

@JorjMcKie I know you can help me with that

**Comment by JorjMcKie** [comment, marked as answer]
Sorry, there is no can do - for principal reasons, has nothing to do with MuPDF / PyMuPDF.
But I **_think_** I know what you are aiming at:

"When I know the position of some text on the page, then where on the image do I find hat text?"

* The page rectangle, and the bboxes of everything on it, are `Rect` rectangles with **_float_** coordinates.
* Every image, and everything on it, is addressable by pixels which has **_integer_** coordinates.
* Rendering a page to an image therefore entails a transformation `page.rect` -> `pix.irect`.

To achieve what you want, you can use the matrix `mat = page.rect.torect(pix.irect)`. Use this matrix to compute the position of some page `bbox` in the image by `bbox * mat`, or even more correctly `(bbox * mat).irect`.
